### PR TITLE
Include rest of Semgrep subcategories in cheatsheet generation

### PR DESCRIPTION
**Pull request by mschwager**
This will give us all the Semgrep features by language, which will help for language graduation.

### CODE REVIEW

File: scripts/generate_test_matrix.py [hunk 1]
```diff
@@ -82,14 +81,16 @@
         "import",
         "typed",
     ],
-    "regexp": ["string"],
+    "regexp": ["string", "fieldname"],
```
Comment (colleend) [inline scripts/generate_test_matrix.py:84]: what does 'fieldname' represent?

Comment (mschwager) [reply scripts/generate_test_matrix.py:84]: Looks like a regexp in an object key: https://github.com/returntocorp/semgrep/blob/develop/semgrep-core/tests/js/regexp_fieldname.sgrep. It's currently only implemented for JS, but I could see it being used in other languages too (e.g. `dict` key in Python).

**Comment by ievans** [comment]
@mschwager I like. Are we updating the build process so that generate cheatsheet is called as part of the build?

**Review by ievans — APPROVED** [review]

**Comment by mschwager** [comment]
> @mschwager I like. Are we updating the build process so that generate cheatsheet is called as part of the build?

Already done! E.g. you can find the "test-matrix" JSON file here: https://github.com/returntocorp/semgrep/actions/runs/368794473

Do we need the HTML file autogenerated as well?

File: scripts/generate_test_matrix.py (hunk 1 above)
Comment (colleend) [reply scripts/generate_test_matrix.py:84]: oh interesting, didn't know that we supported that 😮 , looks good to me!

**Review by colleend — APPROVED** [review]

**Comment by ievans** [comment]
Awesome @mschwager  Nah I don't think so!

**Pull request merged**